src/sentry/integrations/github/client.py

```diff
@@ -102,21 +103,42 @@ def get_tree(self, repo_full_name: str, tree_sha: str) -> JSONData:
 
         return tree
 
-    def get_trees_for_org(self, org_name: str) -> JSONData:
+    def get_trees_for_org(self, gh_org: str, cache_seconds: int = 3600 * 24) -> JSONData:
         """
         This fetches tree representations of all repos for an org.
         """
         trees: JSONData = {}
-        repositories = self.get_repositories(fetch_max_pages=True)
-        # XXX: In order to speed up this function we will need to parallelize this
-        # Use ThreadPoolExecutor; see src/sentry/utils/snuba.py#L358
-        for repo_info in repositories:
-            full_name: str = repo_info["full_name"]
-            branch = repo_info["default_branch"]
-            trees[full_name] = {"default_branch": branch, "files": self.get_tree(full_name, branch)}
+        cache_key = f"githubtrees:repositories:{gh_org}"
```

Comment:
@armenzg While that is true for Github, it is not true for Gitlab which makes me concerned about this approach because I'd imagine we'd want to be able to extend it to Gitlab without major architecture changes.